frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/cells/select_option_cell/extension.dart

```diff
@@ -98,6 +102,36 @@ class SelectOptionTag extends StatelessWidget {
       padding = padding.copyWith(right: 2.0);
     }
 
+    double calculateLuminance(Color color) {
+      List<double> rgb =
+          [color.red, color.green, color.blue].map((c) => c / 255.0).toList();
+
+      for (int i = 0; i < 3; i++) {
+        // Explicitly cast the result of pow to double
+        rgb[i] = rgb[i] <= 0.03928
+            ? rgb[i] / 12.92
+            : pow((rgb[i] + 0.055) / 1.055, 2.4).toDouble();
+      }
+
+      return rgb[0] * 0.2126 + rgb[1] * 0.7152 + rgb[2] * 0.0722;
+    }
+
+    double calculateContrast(Color color1, Color color2) {
+      // Calculate the relative luminance of the colors
+      double luminance1 = calculateLuminance(color1);
+      double luminance2 = calculateLuminance(color2);
+
+      // Ensure luminance1 is the lighter color
+      if (luminance1 < luminance2) {
+        double temp = luminance1;
+        luminance1 = luminance2;
+        luminance2 = temp;
+      }
+
+      // Calculate the contrast ratio
+      return (luminance1 + 0.05) / (luminance2 + 0.05);
+    }
```

Comment:
You should create a class for this, and abstract all magic numbers to constants with good names.